pkg/network/usm/sharedlibraries/ebpf_test.go

```diff
@@ -43,16 +43,20 @@ func (s *EbpfProgramSuite) TestCanInstantiateMultipleTimes() {
 
 	prog := GetEBPFProgram(cfg)
 	require.NotNil(t, prog)
-	t.Cleanup(prog.Stop)
+	t.Cleanup(func() {
+		if prog != nil {
+			prog.Stop()
+		}
+	})
 
 	require.NoError(t, prog.InitWithLibsets(LibsetCrypto))
 	prog.Stop()
+	prog = nil
 
 	prog2 := GetEBPFProgram(cfg)
```

Comment:
At this point variable `prog` is `nil`, why should use new variable prog2 instead of the same `prog`?

Reply:
because of the deferred cleanup

Reply:
`prog` is `nil`, deferred cleanup find condition is false and skips.

Reply:
no it is not. We have two deferred cleanups, one for prog and one for prog2, if you re-use prog, then both deferred cleanups will run on the same variable.
I do encourage you to test it out locally

Reply:
I suggested using one variable and one cleanup procedure.